packages/interactivity/src/vdom.ts

```diff
@@ -141,13 +140,20 @@ export function toVdom( root: Node ): Array< ComponentChild > {
 		if ( directives.length ) {
 			props.__directives = directives.reduce(
 				( obj, [ name, ns, value ] ) => {
-					const directiveMatch = directiveParser.exec( name );
-					if ( directiveMatch === null ) {
+					if ( ! directiveParser.test( name ) ) {
 						warn( `Found malformed directive name: ${ name }.` );
 						return obj;
 					}
-					const prefix = directiveMatch[ 1 ] || '';
-					const suffix = directiveMatch[ 2 ] || 'default';
+
+					const unPrefixed = name.slice( fullPrefix.length );
+					const splitIndex = unPrefixed.indexOf( '--' );
+					const [ prefix, suffix = 'default' ] =
+						splitIndex === -1
+							? [ unPrefixed ]
+							: [
+									unPrefixed.slice( 0, splitIndex ),
+									unPrefixed.slice( splitIndex + 2 ),
+							  ];
```

Comment:
Could we add a couple of comments here so that it's clearer what's going on?

```suggestion
					const [ prefix, suffix = 'default' ] =
						splitIndex === -1
              // If '--' is not found, set 'prefix' to 'unPrefixed'
							? [ unPrefixed ]

              // If '--' is found, split 'unPrefixed' into 'prefix' and 'suffix'
							: [
									unPrefixed.slice( 0, splitIndex ),
									unPrefixed.slice( splitIndex + 2 ),
							  ];
```

Reply:
ok, nevermind, I've just added them in d4739b49b2ad161ba290438fadd453a3caf6b61c